bundle/deploy/terraform/init.go

```diff
@@ -98,14 +109,51 @@ func inheritEnvVars(ctx context.Context, environ map[string]string) error {
 	}
 
 	// Include $TF_CLI_CONFIG_FILE to override terraform provider in development.
-	configFile, ok := env.Lookup(ctx, "TF_CLI_CONFIG_FILE")
+	// See: https://developer.hashicorp.com/terraform/cli/config/config-file#explicit-installation-method-configuration
+	devConfigFile, ok := env.Lookup(ctx, "TF_CLI_CONFIG_FILE")
 	if ok {
+		environ["TF_CLI_CONFIG_FILE"] = devConfigFile
+	}
+
+	// Map $DATABRICKS_TF_CLI_CONFIG_FILE to $TF_CLI_CONFIG_FILE
+	// VSCode extension provides a file with the "provider_installation.filesystem_mirror" configuration.
+	// We only use it if the provider version matches the currently used version,
+	// otherwise terraform will fail to download the right version (even with unrestricted internet access).
+	configFile, err := getEnvVarWithMatchingVersion(ctx, "DATABRICKS_TF_CLI_CONFIG_FILE", "DATABRICKS_TF_PROVIDER_VERSION", schema.ProviderVersion)
+	if err != nil {
+		return err
+	}
+	if configFile != "" {
+		log.Debugf(ctx, "Using Terraform CLI config from DATABRICKS_TF_CLI_CONFIG_FILE at %s", configFile)
 		environ["TF_CLI_CONFIG_FILE"] = configFile
 	}
 
 	return nil
 }
 
+func getEnvVarWithMatchingVersion(ctx context.Context, envVarName string, versionVarName string, currentVersion string) (string, error) {
+	envValue := env.Get(ctx, envVarName)
+	versionValue := env.Get(ctx, versionVarName)
+	if envValue == "" || versionValue == "" {
+		log.Debugf(ctx, "%s and %s aren't defined", envVarName, versionVarName)
+		return "", nil
+	}
+	if versionValue != currentVersion {
+		log.Debugf(ctx, "%s as %s does not match the current version %s, ignoring %s", versionVarName, versionValue, currentVersion, envVarName)
+		return "", nil
+	}
+	_, err := os.Stat(envValue)
+	if err != nil {
+		if os.IsNotExist(err) {
+			log.Debugf(ctx, "%s at %s does not exist, ignoring %s", envVarName, envValue, versionVarName)
+			return "", nil
```

Comment:
Why not a hard error? If someone sets these env vars, they intend to use them. Mistakes are ignored.

Reply:
Environment varialbes in the terminals can have a long timespan, and we can't change them after the initial setup.

One edge case is when users uninstall the extension. The variables will stick in the persistent terminal sessions utill users re-create terminals. Since we don't copy the dependencies and point env vars to the internals of the installed extensions, after the uninstall the variables will point to the place that no longer exist. We can copy dependencies to `.databricks` folder for each project, but I don't think it's worth the hassle

Reply:
Reloading VSCode doesn't restart terminals, they are persistent (at least by default), and so the env vars also persist.

We've not yet made a decision about "Databricks Terminal". We can change the logic here afterwards though